Objects/structseq.c

```
@@ -373,29 +338,55 @@ PyStructSequence_InitType2(PyTypeObject *type, PyStructSequence_Desc *desc)
         k++;
     }
     members[k].name = NULL;
+}
 
+int
+PyStructSequence_InitType2(PyTypeObject *type, PyStructSequence_Desc *desc)
+{
+    PyMemberDef *members;
+    Py_ssize_t n_members, n_unnamed_members;
+
+#ifdef Py_TRACE_REFS
+    /* if the type object was chained, unchain it first
+       before overwriting its storage */
+    if (type->ob_base.ob_base._ob_next) {
+        _Py_ForgetReference((PyObject *)type);
+    }
+#endif
+
+    Py_REFCNT(type) = 0;
```

Setting `Py_REFCNT` worries me. Is there a use case for calling `PyStructSequence_InitType2` on a type structure with `Py_REFCNT != 0`? It seems to me that this should only happen when initializing an already initialized type, i.e. it's always an error. Would it be better to fail instead of reseting the refcount?

---

I was trying to go for preserving what the previous memcpy was doing. That being said, you are right - given that we are already cleaning this up - this should throw an error if this is called twice and this is a good indicator for it!